content/blog/2019-12-compat.md

```diff
@@ -0,0 +1,250 @@
+receiver type for `std::future::Future`'s `poll` method can make migrating
+manual `Future` implementations challenging. Crates such as [`pin-project`] can
+be helpful when dealing with the stack pinning requirement.
+
+In most cases, however, it is often significantly easier to rewrite the existing
+code to use async/await syntax, rather than implementing `Future` by hand or
+using future combinators. Although such a change is larger when measured in
+terms of how many lines of code were modified, the ergonomic benefits of
+async/await syntax can make the migration much easier &mdash; oftentimes, a
+large amount of boilerplate code can simply be removed. Furthermore, this
+has the benefit of resulting in more idiomatic, readable, and maintainable code.
+For most projects, switching to use async/await syntax is the recommended
+migration path.
+
+[`pin-project`]: https://crates.io/crates/pin-project
+
+### Notes
+
+There are a small number of "gotchas" to keep in mind in the current
+`tokio-compat` v0.1. In particular, it is important to note that he
+compatibility thread pool runtime does **not** currently support the
+`tokio` 0.1 [`tokio_threadpool::blocking`][blocking] API. Calls to the
+legacy version of `blocking` made on the compatibility runtime will currently
+fail. In the future, `tokio-compat` will allow transparently replacing
+legacy `blocking` with the `tokio` 0.2 blocking APIs, but in the meantime,
+it will be necessary to convert this code to call into the `tokio` 0.2
+[`task::block_in_place`] and [`task::spawn_blocking`] APIs instead. Since
+[`tokio::fs`] relies on the blocking APIs, the Tokio 0.1 version of `tokio::fs`
+also will not currently work on the compatibility runtime.
+
+Additionally, it's important to keep in mind that Tokio 0.1 and Tokio 0.2
+provide subtly different behavior around spawning tasks and shutting down
+runtimes. In particular, Tokio 0.1 tracks whether a runtime is idle (i.e., it
+has no futures running on it), and provides a `Runtime::shutdown_on_idle` method
+which shuts down the runtime when it becomes idle. On the other hand, Tokio
+0.2's `spawn` functions return `JoinHandle`s, which can be used to await the
+completion of spawned tasks, and users are instead expected to await these
+`JoinHandles` in order to determine when the runtime should shut down.
+Therefore, `tokio-compat` provides both APIs, but it is important to note that
+only tasks spawned _without_ `JoinHandle`s "count" against the runtime being
+considered idle. See [this section][idle] in the documentation for details.
+
+[blocking]: https://docs.rs/tokio-threadpool/0.1.16/tokio_threadpool/fn.blocking.html
+[`task::block_in_place`]: https://docs.rs/tokio/0.2.4/tokio/task/fn.block_in_place.html
+[`task::spawn_blocking`]: https://docs.rs/tokio/0.2.4/tokio/task/fn.spawn_blocking.html
+[`tokio::fs`]: https://docs.rs/tokio/0.1.22/tokio/fs/index.html
+[idle]: TODO_ELIZA_ADD_DOCS_LINK
```

Comment:
This will link to the `tokio-compat` docs but we need to wait until it's actually published first.